### PR TITLE
As a developer, I want sync workflow to preserve commit messages (#95)

### LINKED ISSUE #95: As a developer, I want sync workflow to preserve commit messages so that nabledge repository shows what changed
### Situation

The sync workflow (`.github/workflows/sync-to-nabledge.yml`) currently uses a fixed commit message when pushing to nabledge repository:

```
Sync nabledge marketplace from nabledge-dev
Triggered by: https://github.com/...
```

This makes it impossible to understand what changed in the nabledge repository without clicking through to the original commit.

Additionally, commit messages in nabledge-dev often reference issues using short format (`#95`), which would incorrectly reference nabledge repository issues if copied as-is.

### Pain

Developers reviewing the nabledge repository history cannot see what changed from commit messages alone. They must click through to the trigger URL to understand the actual changes.

Issue references in commit messages point to wrong repository (nabledge instead of nabledge-dev).

### Benefit

Developers can understand what changed in nabledge repository directly from git history, improving visibility and traceability. Issue references correctly link to nabledge-dev issues.

### Success Criteria

- [ ] Sync workflow captures original commit message from nabledge-dev
- [ ] Issue references are converted from `#123` to `nablarch/nabledge-dev#123`
- [ ] Commit message in nabledge repository includes the transformed original message
- [ ] Format preserves both original message and trigger information
- [ ] Works correctly for both single commits and merge commits

### PR DESCRIPTION
**Pull request by kiyotis**
## Approach

Enhanced the sync workflow to preserve original commit messages from nabledge-dev and transform issue references when syncing to nablarch/nabledge repository.

### Key Changes

1. **Commit Message Capture**: Extract original commit subject and body using `git log` with `${{ github.sha }}`

2. **Issue Reference Transformation**: Transform `#123` to `nablarch/nabledge-dev#123` using sed pattern that:
   - Handles issue numbers at start, middle, or end of text
   - Avoids double-transformation of already-qualified references
   - Preserves URLs and version numbers (e.g., `repo#123`, `v1.2.3#456`)

3. **Commit Message Format**:
   - Subject: Transformed original subject
   - Body: Transformed original body + separator + trigger URL
   - Empty body handling included

4. **Error Handling**: Added validation for:
   - Git log command failures (fail fast with clear error messages)
   - Empty commit subjects (validation check)
   - Unique EOF delimiters to prevent heredoc collisions

### Files Modified

- `.github/workflows/sync-to-nabledge.yml`: Added commit message capture and transformation step

### Testing

- Validated sed transformation pattern with 12 test cases (all passed)
- Verified error handling logic
- Confirmed unique EOF delimiter usage

## Tasks

- [x] Capture original commit message from nabledge-dev
- [x] Transform issue references (#123 → nablarch/nabledge-dev#123)
- [x] Combine transformed message with trigger information
- [x] Add error handling for git commands
- [x] Use unique EOF delimiters for heredoc safety
- [x] Test transformation edge cases

## Expert Review

- [DevOps Engineer](https://github.com/nablarch/nabledge-dev/blob/95-sync-commit-messages/.pr/00095/review-by-devops-engineer.md) - Rating: 4/5

**Summary**: Well-implemented workflow changes with good shell scripting practices. Medium-priority improvements for error handling and EOF delimiter safety have been implemented.

## Review Focus

Please verify:
1. Commit message transformation logic is correct
2. Error handling covers expected failure modes
3. Workflow will work correctly in GitHub Actions environment
4. Issue reference transformation handles all edge cases

## Success Criteria

Resolves #95

- [x] Sync workflow captures original commit message from nabledge-dev
- [x] Issue references are converted from `#123` to `nablarch/nabledge-dev#123`
- [x] Commit message in nabledge repository includes the transformed original message
- [x] Format preserves both original message and trigger information
- [x] Works correctly for both single commits and merge commits

### CODE REVIEW

File: .github/workflows/sync-to-nabledge.yml [hunk 1]
```diff
@@ -34,11 +34,57 @@ jobs:
       - name: Validate marketplace structure
         run: .github/scripts/validate-marketplace.sh nabledge-repo
 
+      - name: Capture and transform commit message
+        id: commit_message
+        run: |
```
[the remark this answers is not in the document]
Comment (kiyotis) [reply .github/workflows/sync-to-nabledge.yml]: Fixed

**Commit**: https://github.com/nablarch/nabledge-dev/commit/026be44b0c01bbb1080312cd5192f599f704a89b

Extracted the commit message capture logic to `.github/scripts/capture-commit-message.sh`. The script now accepts the full commit message from `github.event.head_commit.message` context and splits it into subject and body, then transforms issue references.

Co-Authored-By: Claude (jp.anthropic.claude-sonnet-4-5-20250929-v1:0) <noreply@anthropic.com>
